src/core/lib/layer.js

```diff
@@ -537,7 +537,8 @@ export default class Layer {
     if (flags.updateTriggersChanged && !changeFlags.updateTriggersChanged) {
       changeFlags.updateTriggersChanged = flags.updateTriggersChanged;
```

Comment:
Need fixing: The TL;DR `updateTriggersChanged` will have to be merged, not assigned, to support the intended change flags system.

Details:

The idea with `setChangeFlags` is that it will allow us to separate "dirtying" and "updates" of layers, i.e. a layer can gradually accumulate more and more "change flags" until an update is done and clears them all. 

This will for instance allow us to stop updating layers while visible is false, without having to unconditionally regenerate them when visible becomes true again.